### ISSUE TITLE
Scrollbar not taken into account for RTL direction (scrollbar on the left)

### ISSUE THREAD

**Issue by And7s** [popperjs/popper-core#976]
### JSfiddle demo

https://jsfiddle.net/26pu5sjn/

https://i.imgur.com/pjc6yNE.png

### Steps to reproduce the problem

1. have css direction: rtl; overflow: scroll; to the body
2. a scrollbar will be shown on the left side of the screen
3. Add a tooltip with positioning bottom-end;
3. the tooltip will be shifted by the width of the scrollbar to the right.

### What is the expected behaviour?

The tooltip should not be shifted by the right if there is a scrollbar on the left side.

### What went wrong?

The scrollbar on the left side is not taken into the position calculation.

### Any other comments?

Thank you very much for fixing https://github.com/popperjs/popper-core/issues/969 so quickly.

**Comment by FezVrasta** [comment]
I'm wondering if it would be enough to change [this line](https://github.com/popperjs/popper-core/blob/master/src/dom-utils/getDecorations.js#L15) to:

```js
  if (getNodeName(element) === 'html' && getComputedStyle(element).direction !== 'rtl') {
```

or if more work is needed 🤔

**Comment by atomiks** [comment]
It's not the clippingRect calc, rather the core rect calc (`getLayoutRect`). `.offsetLeft` reports 15 for RTL but 0 for LTR... and `left: 0px`  doesn't position it underneath the scrollbar with RTL so the reporting value is wrong 🤦‍♂

**Comment by atomiks** [comment]
Note: this doesn't seem to happen on the main `<body>`, only inside scrolling containers and iframe body/containers it seems.

For those needing a fix now (till we can figure out a clean fix for every case), use a modifier that subtracts `state.rects.popper.x` from `popperOffsets.x`:

```js
const rtlModifier = {
  name: 'rtl',
  enabled: true,
  phase: 'afterRead',
  requires: ['popperOffsets'],
  fn({state}) {
    state.modifiersData.popperOffsets.x -= state.rects.popper.x;
  }
};
```

This only works with `gpuAcceleration` enabled and may have other unconsidered edge cases. Let us know if this works correctly (`2.0.2`+)

**Comment by atomiks** [comment]
Unfortunately, `.clientLeft` reports `0` even when the scrollbar is on the left in the iframe — this needs to be reopened.  It only reports accurately for scrolling containers which is where the patch should always work.

I can't even test this properly because Chrome / FF / Safari all put the scrollbar on the right for `<body>` even when RTL is on; it only switches the scrollbar inside scrolling containers. This means in most cases this shouldn't be a problem unless the scrollbar appears on the left for the `<body>` element.

**It seems one simple fix for RTL is to put `position: relative;` on the `<body>`**. It will position the popper accurately. 

This seems tricky and problematic to fix otherwise 😢

**Comment by And7s** [comment]
Hi, thanks for investigating.
I can still reproduce this issue with version 2.0.4

https://jsfiddle.net/vzn3joLh/1/

![image](https://user-images.githubusercontent.com/5726748/73684217-046ce080-46c4-11ea-8113-1f0049472f2b.png)

expected behaviour: tooltip is shifted to the left so that parent and child align on the right side.

It appears that this issue is happening because of the Iframe embedding. looking at the output on its own is fine, but the scrollbar is only flipped to the left side of the screen in the iframe.

The issue does not appear if the body scrollbar is on the right, but a container has a left side scrollbar:
(here: works as expected)
![image](https://user-images.githubusercontent.com/5726748/73684668-f2d80880-46c4-11ea-8bb6-fc12556df890.png)


https://jsfiddle.net/vzn3joLh/6/

**Comment by atomiks** [comment]
We fixed in #987 but it's not released

**Comment by And7s** [comment]
works perfectly with 2.0.5. thank you.
